docs/content/Modpacks/Materials-and-Elements/Material-Flags.md

```diff
@@ -8,60 +8,218 @@ title: Material Flags
 Using material flags, you can specify several properties of each material, which
 can influence how the material behaves, as well as which items are generated for it.
 
-!!! example "Using material Flags"
+=== "Javascript"
     ```js
     GTCEuStartupEvents.registry('gtceu:material', event => {
         event.create('my_material')
             // ...
             .flags(GTMaterialFlags.FLAMMABLE)
     })
     ```
+=== "Java"
+    ```java
+    public static Material MY_MATERIAL;
+    public static void register() {
+       MY_MATERIAL = new Material.Builder(
+            your_mod_id.id('my_material'))
+            // ...
+            .flags(GTMaterialFlags.FLAMMABLE)
+            .buildAndRegister();
+        }
+    ```
 
 
-## Generate Items
+# Generic Flags
+
+- `NO_UNIFICATION`
+    - Description: Add to material to disable it's unification fully.
+    
+- `DECOMPOSITION_BY_ELECTROLYZING`
+    - Description: Enables electrolyzer decomposition recipe generation (Requires componets to be set).
+
+- `DECOMPOSITION_BY_CENTRIFUGING`
+    - Description: Enables centrifuge decomposition recipe generation (Requires componets to be set).
+
+- `DISABLE_DECOMPOSITION`
+    - Description: Disables decomposition recipe generation for this material.
+
+- `EXPLOSIVE`
+    - Description: Any material with this flag wont have implosion compression recipes, and it will give ash when you arc furnace recycle it instead of that material.
+
+- `FLAMMABLE`
+    - Description: Adding this flag means you cant smelt that material (it wont generate a ebf recipe/furnace recipe) also disables impossion compressor recipes like explosive does.
+
+- `STICKY`
+    - Description: Add to material if it is sticky (Changes the viscosity of the placed fluid. Only the oils and creosote have a placeable state by default).
+
+- `PHOSPHORESCENT`
+    - Description: Adding this flag onto a material gives liquids a luminosity of 15, no matter the fluid state(liquid, gas, plasma) otherwise they default to 10 for specifically liquid state.
+
```

Comment:
```suggestion
- `PHOSPHORESCENT`
    - Description: Adding this flag onto a material gives liquids a luminosity of 15, no matter the fluid state(liquid, gas, plasma). Otherwise they default to 10 for specifically liquid state.

```